test/ruby/test_thread_queue.rb

```diff
@@ -111,6 +111,15 @@ def test_queue_pop_interrupt
     assert_equal(0, q.num_waiting)
   end
 
+  def test_queue_pop_timeout
+    q = Thread::Queue.new
+    t1 = Thread.new { q.pop(timeout: 1) }
+    assert_nil t1.join(2).value
```

Comment:
This one test will take 1 second which is quite long, I'd use 0.1 or even better 0.001 to avoid wasting time during testing.

It would also be nice to test that it does return a pop'd value if there is one.

My bias: of course ruby/spec would be much better for this, and ensure other Ruby implementations have it sooner. For instance there is no way to run MRI tests of a more recent version, but that's supported by ruby/spec with PRETEND_RUBY_VERSION e.g. used in truffleruby, so e.g. it could be added already in TruffleRuby even though it's not aiming for full 3.2 compatibility yet.

Reply:
> This one test will take 1 second which is quite long, I'd use 0.1 or even better 0.001 to avoid wasting time during testing.

I agree, however the reason I test with an integer is that Integer and Float are two distinct codepaths.

> My bias: of course ruby/spec would be much better for this

Yeah, I'll add some specs. I'd like this to be reviewed by someone familiar with the thread scheduler first (likely @ko1), there's no point writing specs if my code is flawed.

Reply:
> there's no point writing specs if my code is flawed.

Just like tests it'd be extra confidence it works as expected, and it'd be useful regardless of the implementation :)

Reply:
I implemented `SizedQueue#push` and added some specs, let me know what you think.

Reply:
> I [...] added some specs, let me know what you think.

Thanks! They look good.